### DISCUSSION TITLE
Connection with mysql

### DISCUSSION THREAD

**Discussion by Alanethm** [Yakifo/amqtt#142] [Q&A] [answered]
hello together,
i have this amqtt during my research of mqtt-Broker writing in python. Really nice project. 

I wanted to ask if the persistence with a mysql-DB is also possible, why i just see a example with sqlite.

i will be great for any reply and any help.

**Comment by FlorianLudwig** [comment]
Hi @Alanethm ,

sorry there never was a reply. Generally speaking aMQTT does not offer any persistence right now. What would you like to persist? Queued messages?

**Comment by ajmirsky** [comment, marked as answer]
session persistence is currently supported with sqlite only. however, the most recent implementation uses sqlalchemy so expanding to support other db backends is a straightforward extension:

https://amqtt.readthedocs.io/en/v0.11.3/plugins/session/

I have opened an issue to track feature development of expanding the existing session plugin to support multiple backends.

**Reply by ajmirsky** [reply]
implemented in PR https://github.com/Yakifo/amqtt/pull/333
